### PR TITLE
Display Posts Widget: Remove overly opinionated CSS

### LINKED ISSUE #1886: Display Posts Widget: remove !important from CSS
We should remove `!important` from the Display Posts Widget CSS to make it easier for folks to style the widget.

Reference:
https://github.com/Automattic/jetpack/blob/3.4.1/modules/widgets/wordpress-post-widget/style.css#L18

Reported here:
https://twitter.com/tsiger/status/582536932372164608

### PR DESCRIPTION
**Pull request by scottsweb**
We have some very opinionated CSS in the remote posts widget at the moment, in #1886 we were asked to remove the `!important` declarations, allowing themes greater control. I have gone a little further to ensure we inherit theme styles as much as possible.

I have done some basic testing the the twenty themes, most of these already supply custom CSS for the widget so there is not a great deal of change to be observed.

We could probably take this further and remove the margin values as well but I am reluctant to make too many changes in one sweep.

Fixes #1886

#### Changes proposed in this Pull Request:

* The CSS has been simplified

#### Testing instructions:

* Add "Display WordPress posts (Jetpack)" widget to a widget area
* Complete the fields, along with a random selection of options
* Observe the look and feel of the widget
* Pull down this PR
* Observe the look and feel of the widget

Are the differences a concern? Does the widget fit better or worse with the existing theme styles?

#### Proposed changelog entry for your changes:

* Display Posts Widget: Remove overly opinionated CSS, please check your themes

### CODE REVIEW

**Review by kraftbj — APPROVED** [review]
Works for me and no obvious issue. Ready to merge once Design approves.

**Comment by MichaelArestad** [comment]
Can you add some before/after screenshots?

**Review by MichaelArestad — APPROVED** [review]
This looks good. I think the odds of this change causing something to break are extremely low.

#### Before
image

#### After
image

**Pull request merged**
